ObolNetwork/charon
# *: release v1.2-rc1

Cherry-picked commits for v1.2-rc1.

**Test command**

- Add cluster lock and definition files to test peers [#3368](https://github.com/ObolNetwork/charon/pull/3368)
- Beacon node simulation [#3361](https://github.com/ObolNetwork/charon/pull/3361)
- General UX [#3370](https://github.com/ObolNetwork/charon/pull/3370)
- Create real blocks with MEV test [#3378](https://github.com/ObolNetwork/charon/pull/3378)
- Version check on beacon tests [#3379](https://github.com/ObolNetwork/charon/pull/3379)
- Rename test performance to test infra [#3380](https://github.com/ObolNetwork/charon/pull/3380)
- Output file improvements [#3384](https://github.com/ObolNetwork/charon/pull/3384)
- Custom number of validators for beacon node simulation [#3385](https://github.com/ObolNetwork/charon/pull/3385)

**Charon exit --all**

- initial refactor [#3248](https://github.com/ObolNetwork/charon/pull/3248)
- add --all flag [#3272](https://github.com/ObolNetwork/charon/pull/3272)
- broadcast all exits [#3288](https://github.com/ObolNetwork/charon/pull/3288)
- fetch all exits [#3291](https://github.com/ObolNetwork/charon/pull/3291)
- enable exit all [#3296](https://github.com/ObolNetwork/charon/pull/3296)
- add custom testnet flags (to enable kurtosis testing) [#3317](https://github.com/ObolNetwork/charon/pull/3317)
- improve logging and error handling [#3347](https://github.com/ObolNetwork/charon/pull/3347)
- increase default Obol API timeout [#3353](https://github.com/ObolNetwork/charon/pull/3353)

**Misc**

- Log leader index [#3334](https://github.com/ObolNetwork/charon/pull/3334)
- Add third Charon relay [#3227](https://github.com/ObolNetwork/charon/pull/3227)
- Fix promrated network overview stats [#3234](https://github.com/ObolNetwork/charon/pull/3234)
- Harden threshold parameter checks [#3242](https://github.com/ObolNetwork/charon/pull/3242), [#3297](https://github.com/ObolNetwork/charon/pull/3297)
- Dependabot to bump only patch versions for our BLS library [#3352](https://github.com/ObolNetwork/charon/pull/3352)
- Optimize Dockerfile [#3281](https://github.com/ObolNetwork/charon/pull/3281)

**Tests / pipelines**

- Fix flaky tests [#3309](https://github.com/ObolNetwork/charon/pull/3309),  [#3316](https://github.com/ObolNetwork/charon/pull/3316), [#3332](https://github.com/ObolNetwork/charon/pull/3332)
- Disable intrange linter [#3282](https://github.com/ObolNetwork/charon/pull/3282)
- Create automate PR for release [#3310](https://github.com/ObolNetwork/charon/pull/3310)
- Use minor versions in pipelines [#3321](https://github.com/ObolNetwork/charon/pull/3321)
- Fix trigger-dispatch for release [#3351](https://github.com/ObolNetwork/charon/pull/3351) [#3381](https://github.com/ObolNetwork/charon/pull/3381)
- Fix linter [#3307](https://github.com/ObolNetwork/charon/pull/3307) (partially)

**Docs**

- Launchpad link broken [#3231](https://github.com/ObolNetwork/charon/pull/3231)
- Docs typos [#3236](https://github.com/ObolNetwork/charon/pull/3236) [#3367](https://github.com/ObolNetwork/charon/pull/3367) [#3369](https://github.com/ObolNetwork/charon/pull/3369)

All of the rest are tens of PRs with simple version bumps across the stack.

category: misc
ticket: none
